common/txmgr/types/head.go

```diff
@@ -0,0 +1,28 @@
+package types
+
+import "github.com/ethereum/go-ethereum/common"
+
+// Head provides access to a chain's head, as needed by the TxManager.
+// This is a generic interface which ALL chains will implement.
+//
+//go:generate mockery --quiet --name Head --output ./mocks/ --case=underscore
+type Head interface {
+	// BlockNumber is the head's block number
+	BlockNumber() int64
+
+	// ChainLength returns the length of the chain followed by recursively looking up parents
+	ChainLength() uint32
+
+	// EarliestInChain traverses through parents until it finds the earliest one
+	EarliestInChain() Head
+
+	// Hash is the head's block hash
+	BlockHash() common.Hash
```

Comment:
now that Head doesn't use generic, how do we specify different hash types for different chains

Reply:
Yes, I didn't touch the Hash yet in this PR.

The current Hash native type(common.Hash) is actually used by Txmgr. So have to see if we can do with a generic, or a common Hash interface.

Reply:
Are other chains just different length arrays? Could we use a byte slice in that case?

Reply:
Can it be an interface with some methods implemented? e.g. `Scan`, `Value`, `Cmp`, `Equal` etc?

Reply:
Yes, I prefer an interface which allows serializing/deserializing, comparing, stringifying etc.

But I don't want to bring that in this PR.

Reply:
why not do this change in the PR? seems easier to put everything together and finalize the interface

Reply:
Just the to and fro we take, for finalizing the approach. Whether to implement Hash as an interface, or byte[], or generic. I want to keep changes isolated more.